onnxruntime/python/onnxruntime_inference_collection.py

```diff
@@ -50,8 +50,8 @@ def check_and_normalize_provider_args(providers, provider_options, available_pro
 
     def set_provider_options(name, options):
         if name not in available_provider_names:
-            raise ValueError("Specified provider '{}' is unavailable. Available providers: '{}'".format(
-                name, ", ".join(available_provider_names)))
+            warnings.warn("Specified provider '{}' is not in available provider names."
```

Comment:
Let's change it to error since we want to let the user know that they need to correct this or else she'll have false expectations.